src/mds/CInode.cc

```diff
@@ -1963,6 +1970,13 @@ void CInode::finish_scatter_update(ScatterLock *lock, CDir *dir,
       
       assert(!dir->is_frozen());
       mut->auth_pin(dir);
+
+      if (lock->get_type() == CEPH_LOCK_INEST && 
+	  !is_auth() && lock->get_next_state() == LOCK_MIX) {
+	dout(10) << "finish_scatter_update finish assimilating dirty rstat on " 
+	  << *dir << dendl; 
+        dir->assimilate_dirty_rstat_inodes_finish(mut, &le->metablob);
```

Comment:
we should execute following lines if rstat changes
 <pre>
mdcache->mds->locker->wrlock_force(&nestlock, mut);
mdcache->mds->locker->mark_updated_scatterlock(&nestlock);
mut->ls->dirty_dirfrag_nest.push_back(&item_dirty_dirfrag_nest);
</pre>

Also need to drop the lock in CInode::_finish_frag_update